modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java

```diff
@@ -424,26 +425,26 @@ public void restoreMessagesFromDeadLetterQueue(@MBeanOperationParameter(name = "
                         break;
                     }
                     AndesMessageMetadata metadata = Andes.getInstance().getMessageMetaData(messageId);
-                    String destination = metadata.getDestination();
-
-                    metadata.setStorageQueueName(AndesUtils.getStorageQueueForDestination(destination,
-                            AMQPUtils.DIRECT_EXCHANGE_NAME,destination,true));
-
-                    messagesToRemove.add(metadata);
-
-                    AndesMessageMetadata clonedMetadata = metadata.shallowCopy(metadata.getMessageID());
-                    AndesMessage andesMessage = new AndesMessage(clonedMetadata);
-
-                    // Update Andes message with all the chunk details
-                    List<AndesMessagePart> messageParts = messageContent.get(messageId);
-                    for (AndesMessagePart messagePart : messageParts) {
-                        andesMessage.addMessagePart(messagePart);
+                    if (null != metadata) {
+                        String destination = metadata.getDestination();
+                        metadata.setStorageQueueName(AndesUtils.getStorageQueueForDestination(destination,
+                                AMQPUtils.DIRECT_EXCHANGE_NAME,destination,true));
+                        messagesToRemove.add(metadata);
+                        AndesMessageMetadata clonedMetadata = metadata.shallowCopy(metadata.getMessageID());
+                        AndesMessage andesMessage = new AndesMessage(clonedMetadata);
+                        // Update Andes message with all the chunk details
+                        List<AndesMessagePart> messageParts = messageContent.get(messageId);
+                        for (AndesMessagePart messagePart : messageParts) {
+                            andesMessage.addMessagePart(messagePart);
+                        }
```

Comment:
Look like these set of code used to created the andesMessage is repeated in the other modified method. Can we use a private method for that.